src/System.Net.Http/tests/FunctionalTests/HttpClientHandlerTest.cs

```diff
@@ -997,6 +1000,143 @@ public static IEnumerable<object[]> VerifyUploadServersStreamsAndExpectedData
             }
         }
 
+        [Fact]
+        public async Task PostAsync_ResponseContentRead_RequestContentDisposedAfterResponseBuffered()
```

Comment:
You should add a test that validates what happens with a `SendAsync(HttpCompletionOption.ResponseHeadersRead)` happens. That should never read the response body at all.